application/src/main/java/org/ehrbase/application/config/SecurityConfig.java

```diff
@@ -30,6 +30,7 @@
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
```

Comment:
For the readability and maintenance, I think that we should create separate configuration classes like : `SecurityConfiguration`, `BasicSecurityConfigucation`, `OAuth2SecurityConfiguration `
And we could enable the correct one using `@ConditionalOnProperty `annotation

Reply:
Good idea! I refactored the config classes to be separated like you suggest. Can you check again if that looks okay?